### PR TITLE
[320] Implement new import strategy based on ecore

### PR DESCRIPTION
**Pull request by GuillaumeEscande**
Re implement the sysml textual import to be based on the ecore Proxy approach and use metamodel resolve methods.

### CODE REVIEW

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java [hunk 1]
```diff
@@ -12,171 +12,62 @@
...
-                this.objectFinder.putEObject(t);
+    public Resource convertResource(InputStream input, ResourceSet editingDomainResourceSet) {
+
+        Resource result = new JSONResourceFactory().createResource(new JSONResourceFactory().createResourceURI(null));
+        editingDomainResourceSet.getResources().add(result);
+        if (input != null) {
+            JsonNode astJson = readAst(input);
+            if (astJson != null) {
+                logger.info("Create the Root eObject containment structure");
+                List<EObject> rootSysmlObjects = AstTreeParser.parseAst(astJson);
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java:46]: Why do you have/call a static method here?
Unless I'm missing something, there is no added value to have a static access here, this should be replace by an instantiation of `AstTreeParser` and a call to `parseAst`.

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java [hunk 2]
```diff
@@ -12,171 +12,62 @@
...
+                result.getContents().addAll(rootSysmlObjects);
+                logger.info("End of create the Root eObject containment structure");
+                logger.info("Try to resolve Imports");
+                AstTreeParser.resolveAllImport(result);
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java:50]: Same remark

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java [hunk 3]
```diff
@@ -12,171 +12,62 @@
...
+                logger.info("End of import resolving");
+                logger.info("Try to resolve all references");
+                AstTreeParser.resolveAllReference(result);
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java:53]: Same remark

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/helper/NameConflictingFilter.java [hunk 4]
```diff
@@ -39,31 +39,35 @@ public class NameConflictingFilter implements Predicate<Membership> {
     @Override
     public boolean test(Membership member) {
         if (member != null) {
-            Element memberElement = member.getMemberElement();
-            if (memberElement != null) {
-                return checkConflictingNames(memberElement);
+            if(member.getMemberName() != null && member.getMemberElement() != null && member.getMemberName() != member.getMemberElement().getName()){
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/helper/NameConflictingFilter.java:42]: formatting is wrong here

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OwningMembershipImpl.java [hunk 5]
```diff
@@ -110,7 +110,7 @@ public Element basicGetOwnedMemberElement() {
         }
         return null;
     }
-
+
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OwningMembershipImpl.java:113]: Please remove this unwanted change

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        List<JsonNode> ownedJson = getOwnedJsonNode(astJson);
+        List<EObject> ownedObject = ownedJson.stream().flatMap(t -> {
+            return AstTreeParser.parseJsonNode(t).stream();
+        }).toList();
+
+        ownedObject.stream().forEach(t -> AstContainmentReferenceParser.ownObject(eObject, t));
+        
+    }
+
+
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java:54]: Please do not let multiple empty lines here and everywhere else in the code.

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    private static List<JsonNode> getOwnedJsonNode(JsonNode astJson) {
+        return ReferenceHelper.extractOwnedObject(astJson);
+    }
+
+    public static void ownObject(EObject owner, EObject owned) {
+        
+        /// Owning Relationship ?s
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java:61]: Please do not let comments like this one, here and everywhere else in the code.

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstWeakReferenceParser.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+                    qualifiedNameTarget = referenceAST.get(AstConstant.TEXT_CONST).asText();
+                }
+
+                EObject reference = AstTreeParser.createObject(referenceAST);
+                
+                if (reference instanceof InternalEObject internalTarget) {
+
+                    internalTarget.eSetProxyURI(URI.createGenericURI("syson-import", "qualifiedName", qualifiedNameTarget));
+
+                    // Faire un switch case pour trouve la eGetReference qui correspond au mot clés referenceType
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstWeakReferenceParser.java:67]: Please do not let comments in french

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    //, AstConstant.HERITAGE_CONST, AstConstant.TYPE_RELATIONSHIPS_CONST, AstConstant.ENDS_CONST, AstConstant.PREFIXES_CONST, 
+
+    // Array 
+    /**
+     * 
+     */
+    private static final List<String> CONTAINMENT_NODES = Arrays.asList(AstConstant.TARGET_CONST, AstConstant.CHILDREN_CONST, AstConstant.HERITAGE_CONST, AstConstant.VALUE_CONST, AstConstant.TARGET_MEMBER_CONST, AstConstant.ASSIGNED_VALUE_CONST, AstConstant.OPERANDS_CONST, AstConstant.EXPRESSION_CONST);
+
+
+    /**
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:42]: Please remove all comments like this one

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+     * 
+     */
+    private static final List<String> NON_CONTAINMENT_NODES = Arrays.asList(AstConstant.TARGET_REF_CONST);
+
+   
+    /**
+     * 
+     * @param astJson
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:50]: Please remove all comments like this one

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+import com.google.common.collect.Streams;
+
+/**
+ * ReferenceHelper.
+ *
+ * @author gescande.
+ */
+public class ReferenceHelper {
+
+    //Node  , AstConstant.TARGET_CHAIN_CONST, , AstConstant.RESULT_CONST, AstConstant.SOURCE_CONST,AstConstant.THEN_CONST, AstConstant.ACCEPTER_CONST, AstConstant.GUARD_CONST, AstConstant.EFFECT_CONST, AstConstant.RECEIVER_CONST, AstConstant.PAYLOAD_CONST,
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:32]: Please remove all comments like this one

Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:32]: Ok

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java (hunk 9 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:42]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java (hunk 10 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/ReferenceHelper.java:50]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstWeakReferenceParser.java (hunk 8 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstWeakReferenceParser.java:67]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java (hunk 7 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java:61]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java (hunk 6 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstContainmentReferenceParser.java:54]: ok

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/helper/NameConflictingFilter.java (hunk 4 above)
Comment (GuillaumeEscande) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/helper/NameConflictingFilter.java:42]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java (hunk 1 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java (hunk 2 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java]: OK

File: backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java (hunk 3 above)
Comment (GuillaumeEscande) [reply backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java]: OK

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OwningMembershipImpl.java (hunk 5 above)
Comment (GuillaumeEscande) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OwningMembershipImpl.java]: ok

**Pull request merged**